src/python_testing/TC_OPCREDS_3_1.py

```diff
@@ -0,0 +1,623 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import copy
+import logging
+import random
+
+import chip.clusters as Clusters
+import chip.discovery as Discovery
+from chip import ChipDeviceCtrl
+from chip.exceptions import ChipStackError
+from chip.interaction_model import InteractionModelError, Status
+from chip.tlv import TLVReader, TLVWriter
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OPCREDS_3_1(MatterBaseTest):
+    def FindAndEstablishPase(self, discriminator: int, setupPinCode: int, nodeid: int, dev_ctrl: ChipDeviceCtrl = None, ):
```

Comment:
nits:
- remove trailing , in argument list
- argument for `discriminator` maybe should be named `longDiscriminator` ?